### PR TITLE
LPS-148809 Delete relationships when an object definition is deleted

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-148809

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @danvb112, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @danvb112, can you please check if these failures are related to your changes? Thanks

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @danvb112, can you please check if these failures are related to your changes? Thanks

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -293,8 +294,14 @@ public ObjectDefinition deleteObjectDefinition(
...
-		// TODO Deleting an object definition should delete related object
-		// relationships
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getObjectRelationships(
+				objectDefinition.getObjectDefinitionId());
+
+		for (ObjectRelationship objectRelationship : objectRelationships) {
+			_objectRelationshipLocalService.deleteObjectRelationship(
+				objectRelationship.getObjectRelationshipId());
+		}
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:304]: @danvb112 we should implement a method called `deleteObjectRelationshipByObjectDefinition` in the objectRelationshipLocalService. I think @p-albuquerque did something similar these days...

Comment (danvb112) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:304]: Hi @gabrielwas, I talked to @p-albuquerque and we concluded that, his solution will also solve this bug.

**Comment by danvb112** [comment]
Resolved by: https://github.com/liferay-objects/liferay-portal/pull/487

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (gabrielwas) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:304]: @danvb112 Thanks for the updates :D

**Pull request closed without merging**